src/transformStateWithClones.js

```diff
@@ -7,7 +7,23 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  let clonedState = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
```

Comment:
Use a destruction for `action`
```suggestion
  for (const action of actions) {
    const { type, .......... } = action;
    
    if (type === 'addProperties') {
```